mobile/src/main/java/ch/epfl/sdp/mobile/ui/social/SocialScreen.kt

```diff
@@ -28,97 +33,135 @@ import ch.epfl.sdp.mobile.ui.social.SocialScreenState.Mode.Searching
+                    SearchResultList(
+                        players = state.searchResult,
+                        onFollowClick = state::onFollowClick,
+                        onShowProfileClick = state::onShowProfileClick,
+                        lazyListState = searchingLazyListState,
+                        key = key,
+                        modifier = Modifier.fillMaxSize(),
+                        contentPadding = totalPadding,
+                    )
+          }
+        }
+      },
+  )
 }
 
 /**
  * Display the list of followed player.
  *
  * @param P the type of the [Person].
  * @param players A list of [Person] that need to be displayed.
- * @param onPersonClick Callback function for click on Item.
+ * @param onShowProfileClick Callback function for click on Item.
  * @param modifier modifier the [Modifier] for the composable.
+ * @param lazyListState the [LazyListState] for the list of items.
+ * @param key a function which uniquely identifies the list items.
+ * @param contentPadding the [PaddingValues] for this list.
  */
+@OptIn(ExperimentalFoundationApi::class)
 @Composable
 fun <P : Person> FollowList(
     players: List<P>,
-    onPersonClick: (P) -> Unit,
-    modifier: Modifier = Modifier
+    onShowProfileClick: (P) -> Unit,
+    modifier: Modifier = Modifier,
+    lazyListState: LazyListState = rememberLazyListState(),
+    key: ((P) -> Any)? = null,
+    contentPadding: PaddingValues = PaddingValues(),
 ) {
   val strings = LocalLocalizedStrings.current
-  Column(modifier) {
-    Text(
-        text = strings.socialFollowingTitle,
-        color = MaterialTheme.colors.primary,
-        style = MaterialTheme.typography.h4,
-        modifier = Modifier.padding(horizontal = 16.dp, vertical = 32.dp))
-    LazyColumn(Modifier.testTag("friendList"), verticalArrangement = Arrangement.spacedBy(16.dp)) {
-      items(players) { friend ->
-        PersonCard(
-            person = friend,
-            modifier = modifier.clickable { onPersonClick(friend) },
-            trailingAction = {
-              OutlinedButton(
-                  onClick = { /*TODO*/},
-                  shape = RoundedCornerShape(24.dp),
-              ) {
-                Text(
-                    modifier = Modifier.padding(horizontal = 8.dp),
-                    text = strings.socialPerformPlay,
-                )
-              }
-            })
-      }
+  LazyColumn(
+      modifier = modifier.testTag("friendList"),
+      verticalArrangement = Arrangement.spacedBy(16.dp),
+      state = lazyListState,
+      contentPadding = contentPadding,
+  ) {
+    item {
+      Text(
+          text = strings.socialFollowingTitle,
+          style = MaterialTheme.typography.h4,
+          modifier = Modifier.padding(horizontal = 16.dp),
+      )
+    }
+    items(
+        items = players,
+        key = key,
+    ) { friend ->
+      PersonItem(
+          modifier = Modifier.clickable { onShowProfileClick(friend) }.animateItemPlacement(),
```

Comment:
[Question] Does changing the order of `clickable` and `animateItemPlacement`  on the `Modifier ` results in a different behavior ?

Reply:
Not in this case. In general, the ordering of `Modifier` may affect the behaviour of the items, however in this case `animateItemPlacement` is implemented as a `ParentDataModifier` so it will not be affected by `clickable {}` (or vice-versa).

`ParentDataModifier` is a way for layout nodes to pass some data to their parent, indicating that they want the parent layout to give them a specific behaviour (in this case, the children nodes want to tell their parent that they expect to be animated) [(source)](https://developer.android.com/reference/kotlin/androidx/compose/ui/layout/ParentDataModifier).